flow/record/adapter/line.py

```diff
@@ -15,23 +18,35 @@ class LineWriter(AbstractWriter):
 
     fp = None
 
-    def __init__(self, path, fields=None, exclude=None, **kwargs):
+    def __init__(self, path, *, fields=None, exclude=None, verbose=False, **kwargs):
         self.fp = open_path_or_stream(path, "wb")
         self.count = 0
         self.fields = fields
         self.exclude = exclude
+        self.verbose = verbose
         if isinstance(self.fields, str):
             self.fields = self.fields.split(",")
         if isinstance(self.exclude, str):
             self.exclude = self.exclude.split(",")
 
     def write(self, rec):
         rdict = rec._asdict(fields=self.fields, exclude=self.exclude)
+        rdict_types = None
+        if self.verbose:
+            rdict_types = {fname: fieldset.typename for fname, fieldset in rec._desc.get_all_fields().items()}
```

Comment:
You could maybe `lru_cache` this on `rec._desc`.

Reply:
nice one, added! including a test to ensure the caching works correctly.